caas/kubernetes/provider/precheck.go

```diff
@@ -39,9 +39,10 @@ func (k *kubernetesClient) PrecheckInstance(ctx context.ProviderCallContext, par
 		return errors.NotSupportedf("constraints %v", strings.Join(unsupported, ","))
 	}
 
-	if params.Series != k8sSeries {
-		return errors.NotValidf("series %q", params.Series)
-	}
+	// TODO(new-charms): handle systems
```

Comment:
TODO(embedded)

for consistency

Can we still check legacy charms though?
We can see if params.Series == k8sSeries or matches the patterm for embedded charms

Reply:
We still need do this kinda check, u see there is a TODO for this.
```go
if params.Series == k8sSeries && meta.Format()>=2{return notValidErr}
```

Reply:
I mean for v1 charms. We are removing the check for those and so could have issues.
And even for v2, we can check the syntax of the series string

Reply:
We discussed and this is ok